integration/query_compat_test.go

```diff
@@ -191,3 +196,24 @@ func TestQueryCompatSort(t *testing.T) {
 
 	testQueryCompat(t, testCases)
 }
+
+func TestQueryCompatBatchSize(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]queryCompatTestCase{
+		"BatchSize1": {
+			filter:    bson.D{},
+			batchSize: 1,
+		},
+		"BatchSize100": {
+			filter:    bson.D{},
+			batchSize: 100,
+		},
+		"BatchSize102": {
+			filter:    bson.D{},
+			batchSize: 102,
+		},
```

Comment:
Let's maybe add a comment about the choice of those numbers?
I see 100 is 1 less than default, and 102 is 1 more than default, I guess it's to test the logic around default value.

Reply:
Added.